### PR TITLE
Sl/105 run refresh

### LINKED ISSUE #105: Run CRD reports incorrect state information when Run fails in DataBricks
- Create a job & run using sample `databricks_v1alpha1_djob.yaml `
- The job will not work as the jar file specified in the spec is invalid. 
- Try and create a Run object for this job using the `databricks_v1alpha1_run_job.yaml`
- The run will fail because the jar file is invalid, but the Operator throws an exception and never updates its state.
- Log shows the following error:
```
2019-11-12T14:36:56.612Z        ERROR   controller-runtime.controller   Reconciler error        {"controller": "run", "request": "default/run-sample", "error": "error when refreshing run: Run result unavailable: job failed with error message\n Library installation failed for library jar: \"dbfs:/my-jar.jar\"\n. Error messages:\njava.lang.Throwable: java.io.FileNotFoundException: dbfs:/my-jar.jar"}
github.com/go-logr/zapr.(*zapLogger).Error
        /go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0-beta.4/pkg/internal/controller/controller.go:218
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0-beta.4/pkg/internal/controller/controller.go:192
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker
        /go/pkg/mod/sigs.k8s.io/controller-runtime@v0.2.0-beta.4/pkg/internal/controller/controller.go:171
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:152
k8s.io/apimachinery/pkg/util/wait.JitterUntil
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:153
k8s.io/apimachinery/pkg/util/wait.Until
        /go/pkg/mod/k8s.io/apimachinery@v0.0.0-20190404173353-6a84e37a896d/pkg/util/wait/wait.go:88
```

- This error then proceeds to keep recurring in the logs

Expected behaviour:
- No error reported
- Correct status reported

### PR DESCRIPTION
**Pull request by stuartleeks**
Fixes #105 

To address the issue with not being able to reconcile runs(#105) this PR removes the check that propagated an error that was reported as part of the run state from the API to be a reconciler error. WIth this PR the reconciler now completes successfully and the run status is reported in the status for the run CRD instance (and availble via `kubectl describe run myrun`

### CODE REVIEW

**Comment by Azadehkhojandi** [comment]
Can you please cherry-pick your changes related to #105 and send PR separately?

File: controllers/run_controller_databricks.go [hunk 1]
```diff
@@ -104,9 +103,6 @@ func (r *RunReconciler) refresh(instance *databricksv1alpha1.Run) error {
 	if err != nil {
 		return err
 	}
-	if len(runOutput.Error) > 0 {
```
Comment (Azadehkhojandi) [inline controllers/run_controller_databricks.go]: Let's catch up about this issue

**Comment by stuartleeks** [comment]
I've rebased this on master and re-tested following the [steps outlined in the PR](https://github.com/microsoft/azure-databricks-operator/issues/105#issuecomment-560453348)

```bash
$ kubectl get run
NAME         AGE     RUNID   STATE
run-sample   2m23s   15      PENDING
```


After a while, the job is in the error state:

```bash
export JOB_ID=15 # taken from kubectl get run output above
curl -H "Authorization: Bearer $DATABRICKS_TOKEN" $DATABRICKS_HOST/api/2.0/jobs/runs/get?run_id=$JOB_ID
```

```json
{
  "job_id": 24,
  "run_id": 15,
  "number_in_job": 1,
  "state": {
    "life_cycle_state": "INTERNAL_ERROR",
    "result_state": "FAILED",
    "state_message": "Library installation failed for library jar: \"dbfs:/my-jar.jar\"\n. Error messages:\njava.lang.Throwable: java.io.FileNotFoundException: dbfs:/my-jar.jar"
  },
// rest omitted for brevity
```

Once the operator has next refreshed the `INTERNAL_ERROR` status is reflected in the run status:

```bash
$ kubectl get run
NAME         AGE   RUNID   STATE
run-sample   10m   15      INTERNAL_ERROR
```

Observing the operator logs shows that the status is still being reconciled every 30 seconds.

**Pull request merged**